.changeset/six-nails-destroy.md

```diff
@@ -0,0 +1,19 @@
+---
+'@backstage/plugin-vault-backend': minor
+---
+
+Support Kubernetes authentication for the Vault plugin. To do that, modify the current `vault.token` string
+to an object like the following:
+
+```yaml
+vault:
+  token:
+    type: kubernetes
+    role: <KUBERNETES_ROLE>
+    authPath: <KUBERNETES_AUTH_PATH> # Optional. It defaults to 'kubernetes', but you could set a different authPath if needed
+    serviceAccountTokenPath: <PATH_TO_JWT> # Optional. It defaults to '/var/run/secrets/kubernetes.io/serviceaccount/token'. Where the JWT token is located
+```
+
+To make sure the Kubernetes authentication is applied, and you're using the old
+backend setup, you will need to call `await builder.loadToken()` on demand, to make sure
+the token is fetched. And _always_ after the `builder.build()`.
```

Comment:
> However, another possible solution could be adding the call to renew the token automatically during the plugin creation, so the devs don't need to worry about this method any longer

but the token can expire so that won't completely solve the problem, right?

Reply:
Yes, that's true.With the current approach, any request would fail until the reloadToken is called. If we make the call on each apiCall, this should never happen.

One possible idea is storing more info about the token, such as when it's expired. And make a call when the time got exceeded

Reply:
> One possible idea is storing more info about the token, such as when it's expired. And make a call when the time got exceeded

yeah exactly I think it's a great idea! I was thinking the same with the `loadOrRenewToken()` function above 👆

Reply:
> yeah exactly I think it's a great idea! I was thinking the same with the loadOrRenewToken() function above 👆
I'm currently trying this out. I have a method that is doing this already. However, I'm afraid the `renewToken` method would probably break if the token gets expired, because it cannot renew itself if the token is expired. Even if we reduce the expiration time to leave a threshold and force the refresh before this actually happens.
But since it would be called from the `callApi`, if no secrets are requested from the UI, then it might get expired anyways. One example: when nobody is requesting info about vault secrets during the night